### PR TITLE
CMCL-0000: Fix deprecated calls in 2023

### PR DESCRIPTION
**Pull request by gaborkb**
### Purpose of this PR
Replaced some deprecated calls with their new version in 2023+.

### Testing status
- [ ] Added an automated test
- [ ] Passed all automated tests
- [x] Manually tested

### CODE REVIEW

File: com.unity.cinemachine/Editor/Menus/CinemachineMenu.cs [hunk 1]
```diff
@@ -305,9 +305,13 @@ static CinemachineBrain GetOrCreateBrain()
                 return CinemachineCore.Instance.GetActiveBrain(0);
 
             // Create a CinemachineBrain on the main camera
-            Camera cam = Camera.main;
+            var cam = Camera.main;
             if (cam == null)
+#if UNITY_2023_1_OR_NEWER
+                cam = Object.FindAnyObjectByType<Camera>(FindObjectsInactive.Exclude);
```
Comment (gaborkb) [inline com.unity.cinemachine/Editor/Menus/CinemachineMenu.cs:311]: The other option would be to call FindFirstObjectOfType. This is slower, but that's what the deprecated call `FindObjectOfType()` did. 

`FindAnyObjectByType` is faster than `FindFirstObjectOfType`, but if more cameras are in the scene, `FindAnyObjectByType` may not find the same on subsequent calls. 

I think we don't need consistency in this case, so I chose the faster method.

**Review by AntoineCharton — APPROVED** [review]
Looks good 👍

File: com.unity.cinemachine/Editor/Menus/CinemachineMenu.cs (hunk 1 above)
Comment (glabute) [reply com.unity.cinemachine/Editor/Menus/CinemachineMenu.cs:311]: I think the slower version is better in this case.  We do want consistency, or we may end up creating multiple Brains if Camera.main is never set.

**Review by glabute — CHANGES_REQUESTED** [review]
Note that a backport to 2.9 will be necessary.  2.9 must support 2023 also.

File: com.unity.cinemachine/Runtime/Core/CinemachineVirtualCameraBase.cs [hunk 2]
```diff
@@ -801,7 +801,12 @@ class OnDomainReload
         { 
             static OnDomainReload() 
             {
+#if UNITY_2023_1_OR_NEWER
+                var vcams = FindObjectsByType<CinemachineVirtualCameraBase>
+                    (FindObjectsInactive.Include, FindObjectsSortMode.None);
```
Comment (glabute) [inline com.unity.cinemachine/Runtime/Core/CinemachineVirtualCameraBase.cs:806]: should use FindFirstObjectOfType here also

File: com.unity.cinemachine/Runtime/Core/CinemachineVirtualCameraBase.cs [hunk 3]
```diff
@@ -801,7 +801,12 @@ class OnDomainReload
         { 
             static OnDomainReload() 
             {
+#if UNITY_2023_1_OR_NEWER
+                var vcams = FindObjectsByType<CinemachineVirtualCameraBase>
+                    (FindObjectsInactive.Include, FindObjectsSortMode.InstanceID);
```
Comment (glabute) [inline com.unity.cinemachine/Runtime/Core/CinemachineVirtualCameraBase.cs:806]: Sorry, I was wrong to say earlier that the order needs to be determinate here.  It doesn't.  Better to use the unsorted version.

**Review by glabute — CHANGES_REQUESTED** [review]

**Review by glabute — APPROVED** [review]

**Pull request merged**